scripts/issue/554/upgrade_bveOXD_LP.py

```diff
@@ -0,0 +1,48 @@
+from brownie import interface, Contract
+from rich.console import Console
+from brownie.test.managers.runner import RevertContextManager as reverts
+from helpers.addresses import registry
+from great_ape_safe import GreatApeSafe
+
+C = Console()
+
+STRAT_ADDRESS = registry.ftm.strategies["native.bveOXD-OXD"]
+DEV_PROXY = registry.ftm.badger_wallets.devProxyAdmin
+DEV_MULTI = registry.ftm.badger_wallets.dev_multisig
+NEW_LOGIC = registry.ftm.logic["StrategybveOxdOxdStakingOptimizer"] 
+
+def main(simulation="false"):
+    safe = GreatApeSafe(DEV_MULTI)
+    strat_contract = Contract(STRAT_ADDRESS)
+    strat = interface.IStrategy(STRAT_ADDRESS, owner=safe.account)
+
+    # Record storage variables
+    attributes = {}
+    for attr in strat_contract.signatures:
+        try:
+            attributes[attr] = getattr(strat_contract, attr).call()
+        except:
+            C.print(f"[red]Error storing {attr}[/red]")
+            pass
+
+    # Test failing harvest
+    if simulation == "true":
+        with reverts("BaseV1Router: INSUFFICIENT_B_AMOUNT"):
+            strat.harvest({"from": safe.account})
+
+    # Execute upgrade
+    proxyAdmin = interface.IProxyAdmin(DEV_PROXY, owner=safe.account)
+    proxyAdmin.upgrade(strat.address, NEW_LOGIC)
+
+    # Confirm storage variables
+    for attr in strat_contract.signatures:
```

Comment:
should we make this a function somewhere? like a generalized check_storage(contract)? and also think we should fail if one of the attributes isn't equal so maybe have something like following

```
is_storage_wrong = False
for attr in attrs:
   try:
      whatever
   except:
      is_storage_wrong = True

if is_storage_wrong:
    fail 
```

Reply:
Hmm should we at least log which ones pass and which ones fail so we can at least manually visually confirm everything looks good?

Reply:
Btw, this isn't an exhaustive check since it only verifies public value-type storage variables and public view functions

Reply:
That's right, it has limitations. Not sure if this approach could be generalized efficiently.